src/lib.rs

```diff
@@ -327,6 +327,20 @@ impl TransactionNote {
             TransactionNote::Freeze(note) => note.proof.clone(),
         }
     }
+
+    /// Compute amount of claimable transaction fee
+    pub fn calculate_fee(txns: &[Self]) -> Result<u64, TxnApiError> {
```

Comment:
personally this feels strange to be `TransactionNote::calculate_fee()` than just a standalone function as it is now. Because `calculate_fee` is not a property of a *single* note, but a list of them, so shouldn't be a method for this struct imo.